deps/common/log/log.cpp

```diff
@@ -312,6 +312,18 @@ void Log::set_context_getter(std::function<intptr_t()> context_getter)
 
 intptr_t Log::context_id() { return context_getter_(); }
 
+void Log::get_time_lock()
+{
+  pthread_mutex_init(&get_time_lock_, NULL);
```

Comment:
1. if you init a lock every time, then the lock cannot protect the critical section;
2. get_time_lock maybe called in parallel, and `pthread_mutex_init` is not thread-safe;
3. std::mutex is easier.